### PR TITLE
Implements `DismissJobError` GQL mutation

### PR DESCRIPTION
**Pull request by vyzaldysanchez**
Closes: https://app.shortcut.com/chainlinklabs/story/21721/add-dismissjoberrors-mutation

Aims to enable a GQL mutation to dismiss a single job error

### CODE REVIEW

File: core/web/pipeline_job_spec_errors_controller.go [hunk 1]
```diff
@@ -19,7 +20,7 @@ type PipelineJobSpecErrorsController struct {
 // Destroy deletes a PipelineJobSpecError record from the database, effectively
 // silencing the error notification
 func (psec *PipelineJobSpecErrorsController) Destroy(c *gin.Context) {
-	jobSpec := job.Job{}
+	jobSpec := job.SpecError{}
```
Comment (jkongie) [inline core/web/pipeline_job_spec_errors_controller.go:23]: Was this broken or just badly named?

Comment (vyzaldysanchez) [reply core/web/pipeline_job_spec_errors_controller.go:23]: It was referring to the wrong struct and as a result forcing the method that receives the ID to be int32 when the actual ID is int64.

**Comment by cl-homu** [comment]
:umbrella: The latest upstream changes (presumably 3f7669cb1df815647bff5a15446c652fe714e605) made this pull request unmergeable. Please [resolve the merge conflicts](https://rustc-dev-guide.rust-lang.org/git.html#rebasing-and-conflicts).

File: core/web/resolver/job_error.go [hunk 2]
```diff
@@ -47,3 +46,36 @@ func (r *JobErrorResolver) Occurrences() int32 {
...
+
+// -- DismissJobError Mutation --
+
+type DismissJobErrorPayloadResolver struct {
+	specError *job.SpecError
+	NotFoundErrorUnionType
+}
+
+func NewDismissJobErrorPayload(specError *job.SpecError, err error) *DismissJobErrorPayloadResolver {
+	e := NotFoundErrorUnionType{err: err, message: "PipelineJobSpecError not found"}
```
Comment (jkongie) [inline core/web/resolver/job_error.go:58]:
```suggestion
	e := NotFoundErrorUnionType{err: err, message: "JobSpecError not found"}
```

**Review by jkongie — CHANGES_REQUESTED** [review]

**Comment by cl-homu** [comment]
:umbrella: The latest upstream changes (presumably 5deb066ba202f6bc765785781e456e92508890fb) made this pull request unmergeable. Please [resolve the merge conflicts](https://rustc-dev-guide.rust-lang.org/git.html#rebasing-and-conflicts).

**Review by jkongie — APPROVED** [review]

**Pull request merged**